tests/k8s_test.go

```diff
@@ -29,72 +29,81 @@ import (
 	"os"
 	"testing"
 
-	"github.com/skydive-project/skydive/gremlin"
+	g "github.com/skydive-project/skydive/gremlin"
 	"github.com/skydive-project/skydive/tests/helper"
 )
 
 func k8sConfigFile(name string) string {
 	return "./k8s/" + name + ".yaml"
 }
 
-const (
-	manager    = "k8s"
-	objectName = "skydive-test"
-)
+func k8sObjectName(name string) string {
+	prefix := "skydive-test"
+	if name != "" {
+		return prefix + "-" + name
+	}
+	return prefix
+}
 
-var (
-	networkPolicyConfig = k8sConfigFile("networkpolicy")
-	namespaceConfig     = k8sConfigFile("namespace")
+const (
+	manager = "k8s"
 )
 
 var (
 	nodeName, _       = os.Hostname()
-	podName           = objectName
-	containerName     = objectName
-	networkPolicyName = objectName
-	namespaceName     = objectName
+	podName           = k8sObjectName("pod")
+	containerName     = k8sObjectName("container")
+	networkPolicyName = k8sObjectName("")
+	namespaceName     = k8sObjectName("")
 )
 
-var (
-	setupPod = []helper.Cmd{
-		{"kubectl run " + podName +
+func setupFromDeploymnet(name string) []helper.Cmd {
+	return []helper.Cmd{
+		{"kubectl run " + k8sObjectName(name) +
 			"  --image=gcr.io/google_containers/echoserver:1.4" +
 			"  --port=8080", true},
+		{"sleep 2", true},
 	}
-	tearDownPod = []helper.Cmd{
-		{"kubectl delete deployment " + podName, false},
-	}
-	setupNetworkPolicy = []helper.Cmd{
-		{"kubectl create -f " + networkPolicyConfig, true},
-	}
-	tearDownNetworkPolicy = []helper.Cmd{
-		{"kubectl delete -f " + networkPolicyConfig, false},
+}
+
+func tearDownFromDeployment(name string) []helper.Cmd {
+	return []helper.Cmd{
+		{"kubectl delete deployment " + k8sObjectName(name), false},
+		{"sleep 2", true},
```

Comment:
I would prefer if we tested that the deployment/pod/networkpolicy was successfully deleted instead of waiting 2 seconds :-/

Reply:
ok, have addressed

Reply:
I was suggesting to check that the resource was successfully removed from the graph. That's ok, we'll address this in an other PR